src/currency/currency-formatter.ts

```diff
@@ -0,0 +1,72 @@
+import { CurrencyConfig } from './currency';
+
+export default class CurrencyFormatter {
+    private _decimalPlaces: number;
+    private _symbolLocation: string;
+    private _symbol: string;
+    private _thousandsSeparator: string;
+    private _decimalSeparator: string;
+
+    constructor(
+        currencySettings?: CurrencyConfig | any
```

Comment:
`currencySettings` seems to be a required property, as the class will throw an error if it is missing. If you want to make the property optional, you should provide some default values.